### PR TITLE
DISPATCH-1411 - Added UTC datetime and router id to all options of qd…

### PR DESCRIPTION
**Pull request by ganeshmurthy**
…stat

### CODE REVIEW

**Comment by ganeshmurthy** [comment]
Added time in UTC and router id to qdstat options. Earlier, only qdstat --all-entities or qdstat --all-entities --all-routers showed the time and router id.

File: tools/qdstat.in [hunk 1]
```diff
@@ -155,7 +155,19 @@ class BusManager(Node):
             return text[:-1]
         return text
 
-    def displayEdges(self):
+    def displayRouterId(self):
+        objects = self.query('org.apache.qpid.dispatch.router')
+        router = objects[0]
+        if router:
+            print (router.id)
```
Comment (kgiusti) [inline tools/qdstat.in]: Please remove the space between the print and it's argument list.
This is not PEP8 compliant and will cause python lint checkers to flag it as a violation.

File: tools/qdstat.in [hunk 2]
```diff
@@ -155,7 +155,19 @@ class BusManager(Node):
...
+
+    def display_datetime_router_id(self):
+        print (str(datetime.utcnow()) + " UTC")
+        self.displayRouterId()
+        print ("")
```
Comment (kgiusti) [inline tools/qdstat.in]: ditto

File: tools/qdstat.in [hunk 3]
```diff
@@ -574,6 +607,14 @@ class BusManager(Node):
 
         objects = self.query('org.apache.qpid.dispatch.router.config.autoLink', cols, limit=self.opts.limit)
 
+        if show_date_id:
+            self.display_datetime_router_id()
+
+        if not objects:
+            print ("AutoLinks")
```
Comment (kgiusti) [inline tools/qdstat.in]: ditto

File: tools/qdstat.in [hunk 4]
```diff
@@ -574,6 +607,14 @@ class BusManager(Node):
...
+            print ("No AutoLinks found")
```
Comment (kgiusti) [inline tools/qdstat.in]: here also

File: tools/qdstat.in [hunk 5]
```diff
@@ -602,6 +644,17 @@ class BusManager(Node):
         link_routes = self.query('org.apache.qpid.dispatch.router.config.linkRoute', cols, limit=self.opts.limit)
 
         have_add_del_prefix = False
+
+        if not link_routes:
+            if show_date_id:
+                self.display_datetime_router_id()
+            print ("Link Routes")
```
Comment (kgiusti) [inline tools/qdstat.in]: and here

File: tools/qdstat.in [hunk 6]
```diff
@@ -602,6 +644,17 @@ class BusManager(Node):
...
+            print ("No Link Routes found")
```
Comment (kgiusti) [inline tools/qdstat.in]: here also

File: tools/qdstat.in [hunk 7]
```diff
@@ -667,14 +723,27 @@ class BusManager(Node):
...
-        self.displayAddresses()
-        self.displayConnections()
-        self.displayAutolinks()
-        self.displayLinkRoutes()
-        self.displayGeneral()
-        self.displayMemory()
+    def show_all(self, show_summary=True):
+        if show_summary:
+            # print the datetime and router id
+            print (str(datetime.utcnow())  + " UTC")
```
Comment (kgiusti) [inline tools/qdstat.in]: don't forget here!

File: tools/qdstat.in [hunk 8]
```diff
@@ -667,14 +723,27 @@ class BusManager(Node):
...
+            self.displayRouterId()
+            print ("")
```
Comment (kgiusti) [inline tools/qdstat.in]: and here!

File: tools/qdstat.in [hunk 9]
```diff
@@ -667,14 +723,27 @@ class BusManager(Node):
...
+
+        self.displayRouterLinks(show_date_id=False)
+        print ("")
```
Comment (kgiusti) [inline tools/qdstat.in]: woop!  Lotsa spaces after 'print' here!

File: tools/qdstat.in [hunk 10]
```diff
@@ -687,49 +756,59 @@ class BusManager(Node):
...
             for node in nodes:
+                # We already have more than one node, just print the
+                # router id here instead of asking the router for it.
                 super(BusManager, self).set_client(node[6:])
                 parts = node.split("/")
-                print ("Router ", parts[3])
-                self.show_all()
+                print (str(datetime.utcnow()) + " UTC")
+                print(parts[3])
+                print ("")
```
Comment (kgiusti) [inline tools/qdstat.in]: nope

File: tools/qdstat.in [hunk 11]
```diff
@@ -687,49 +756,59 @@ class BusManager(Node):
...
+                print("")
                 print("")
         else:
             self.show_all()
 
     def displayMain(self, identitys, opts):
         main = opts.show
         if opts.all_routers and main:
-            print (str(datetime.now()))
+            print (str(datetime.utcnow()) + " UTC")
```
Comment (kgiusti) [inline tools/qdstat.in]: and here

File: tools/qdstat.in [hunk 12]
```diff
@@ -687,49 +756,59 @@ class BusManager(Node):
...
             has_nodes, nodes = self.has_nodes()
             if has_nodes:
+                # We already have more than one node, just print the
+                # router id here instead of asking the router for it.
                 for node in nodes:
                     super(BusManager, self).set_client(node[6:])
                     parts = node.split("/")
-                    print ("Router ", parts[3])
-                    self.show_entity(main)
+                    print (parts[3])
```
Comment (kgiusti) [inline tools/qdstat.in]: and here too

**Pull request closed without merging**